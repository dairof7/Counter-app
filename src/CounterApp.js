//rafcp      con este sale todo el componente
import React, { useState } from 'react'
import PropTypes from 'prop-types'

const CounterApp = ({ num=10 }) => {


    const [ counter, setCounter ] = useState( num );  //aqui se retorna un arreglo [valor, funcion]

    //esta funciona va a manejar el evento
    const handleAdd = () => {
        setCounter(counter + 1)
        //otra forma de hacerlo es con una funcion
        //setCounter( (c) => c + 1 )
    }

        const handleSub = () => {
            setCounter(counter - 1)
        }
        const handleReset = () => {
            setCounter(num)
        }
    return (
        <>
            <h1> CounterApp </h1>
            <h2> { counter } </h2>
            
            {/* <button onClick={ (e) => { handleAdd (e) } }>Incrementar</button> */}
            {/* esto se puede reescribir asi y manejarlo con la funcion 
            el argumento 'e' que viene del evento se pasa a la funcion handleAdd */}
            <button onClick={ handleAdd } >Incrementar</button>
            <button onClick={ handleReset } >Reset</button>
            <button onClick={ handleSub } >Decrementar</button>

            {/* si se pone la funcion de esta manera () se va a ejecutar antes del
            evento es decir cuando ejecuta el codigo por primera vez
            <button onClick={ handleAdd() } >Incrementar</button> */}


        </>
    )
}
CounterApp.propTypes = {
num: PropTypes.number
}

export default CounterApp

