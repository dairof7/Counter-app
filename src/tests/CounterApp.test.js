import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom'
import CounterApp from '../CounterApp';

describe('Prueba CounterApp componente', () => {


    //no es una buena practica poner el shallow......
    //pero se hace para no perder el asistente de visual
    let wrapper = shallow( <CounterApp /> );
    //esta funcion se ejecuta antes de cada prueba
    //rehacemos el wrapper cada vez
    beforeEach( () => {
        wrapper = shallow( <CounterApp /> );
    });

    test('debe mostrar CounterApp correctamete', () => {
        expect ( wrapper ).toMatchSnapshot();
    })
    
    test('debe mostrar CounterApp correctamete con valor de 100', () => {
        const valor = 100;
        const wrapper = shallow( <CounterApp num={ valor } /> )
        //trim borra los espacios alrededor de la variable
        const contador = wrapper.find('h2').text().trim()
        console.log(contador)
        expect (contador).toBe( String(valor) )
    })
    
    test('debe de incrementar con el boton incrementar', () => {
        //at me permite seleccionar el primer elemento encontrado
        wrapper.find('button').at(0).simulate('click');
        const contador = wrapper.find('h2').text().trim()
        expect (contador).toBe( '11' )
    })

    test('debe de decrementar con el boton decrementar', () => {
        //at me permite seleccionar el primer elemento encontrado
        wrapper.find('button').at(2).simulate('click');
        const contador = wrapper.find('h2').text().trim()
        expect (contador).toBe( '9' )
    })

    test('debe de colocar el valor por defecto del reset ', () => {
        const valor = 105;
        const wrapper = shallow( <CounterApp num={ valor } /> )
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        const contador = wrapper.find('h2').text().trim()
        expect (contador).toBe( '105' )
        
        
    })
    
    
})
