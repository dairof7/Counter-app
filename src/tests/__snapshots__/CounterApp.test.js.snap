// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba CounterApp componente debe mostrar CounterApp correctamete 1`] = `
<Fragment>
  <h1>
     CounterApp 
  </h1>
  <h2>
     
    10
     
  </h2>
  <button
    onClick={[Function]}
  >
    Incrementar
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    Decrementar
  </button>
</Fragment>
`;
